### ISSUE TITLE
language server attempts to parse binary files

### ISSUE THREAD

**Issue by haberdashPI** [JuliaEditorSupport/julia-vscode#669]
Vim leaves an `.undo` folder around in my projects with files named things like `.undo/%Users%davidlittle%.julia%v0.6%Cairo%deps%build.jl`. Despite its name, this file is not a text file, but a binary, saved undo history of build.jl.

When the vscode extension opens in said project I get the following error:

```julia
[ Info: StaticLint store set
[ Info: parsed /Users/davidlittle/googledrive/Work/Software/bistable/.undo/%Users%davidlittle%.julia%v0.6%Cairo%deps%build.jl
[ Info: Language Server crashed with
[ Info: Base.InvalidCharError{Char}('\x9f')
```

An obvious solution to this problem is for julia-vscode to respect `.gitignore`. I have listed `.unod/*` in my `.gitignore` file, so it is surprising to me that the language server would even try to read this file. More generally, seems like it would be useful to have the language server fail gracefully when a file turns out to be a binary file, not a text file. 

Happy to help fix this issue if you can point me in the right direction in the code base as to where I could address this issue.

**Comment by davidanthoff** [comment]
I don't think we should look at ``.gitignore``, whether a file is excluded from git is really orthogonal to whether it should be parsed or not.

BUT, we should most certainly not crash!  I could also imagine that we add some config option to ``.vscode`` or something like that. I'm not sure what the VS Code standard config file situation looks like right now, but I think that is where we should probably hook in?

Or maybe have a custom config file for julia that can be used to exclude files from linting?

**Comment by haberdashPI** [comment]
Aye, yeah, I guess it makes sense to make them different.

You could also add another setting; that could be a list of file patterns to not parse. For example:

```json
{
    "julia.lsIgnore": [
       ".undo/**"
    ]
}
```

That way, you can specify for all projects (in user settings), one workspace (in workspace settings) or per folder (in folder settings). 

On the other hand, it now occurs to me that maybe this is really an issue for the [langauge server](https://github.com/JuliaEditorSupport/LanguageServer.jl). So that's more in line with your idea of having a file named `.julials` or something that's (perhaps) a TOML file with an option to exclude certain file patterns from parsing.
